blockstore/apps/bundles/storage.py

```diff
@@ -90,12 +90,18 @@ def __init__(self):
             raise self.BackendNotAvailable from attr_error
         if not (key and secret):
             raise self.BackendNotAvailable
-        self.s3_backend = S3Boto3Storage(
-            # All other S3 settings will be pulled in automatically from Django settings
-            # (such as AWS_QUERYSTRING_EXPIRE and AWS_STORAGE_BUCKET_NAME).
-            access_key=key, secret_key=secret
+
+        # Provide the bucket name and prefix if configured;
+        # All other S3 settings will be pulled in automatically from Django settings
+        # (such as AWS_QUERYSTRING_EXPIRE and AWS_STORAGE_BUCKET_NAME).
+        s3_backend_args = dict(
+            access_key=key,
+            secret_key=secret,
+            **settings.BUNDLE_ASSET_STORAGE_SETTINGS['STORAGE_KWARGS']
         )
 
+        self.s3_backend = S3Boto3Storage(**s3_backend_args)
+
```

Comment:
@symbolist  or @kdmccormick -- can you help me understand why the `AssetStorage` class uses the `url_backend` for the url generation, but then uses `default_storage` for the other storage calls? What is the use case for having the storage backend which generates the URLs to be different from the one that actually saves/deletes the files?

I'm trying to figure out why I have to set both `BUNDLE_ASSET_URL_STORAGE_KEY` + `AWS_ACCESS_KEY_ID` + all of the other storage variables, and also `DEFAULT_FILE_STORAGE: storages.backends.s3boto3.S3Boto3Storage` in order to get the blockstore to save its media files to s3. It seems like we only really need the `BUNDLE_ASSET_*` settings (which could default to the `AWS_*` ones if unset).

Reply:
This change was introduced in this PR: https://github.com/openedx/blockstore/pull/101. I hope the description is helpful?

Reply:
It is, thank you @symbolist !

But in order to run blockstore as an app inside edx-platform, we're going to have to remove the usage of `default_storage` as configured by the `DEFAULT_FILE_STORAGE` and `AWS_*` access variables. I'll work on updating the Blockstore IDA's settings.production to use these app-level defaults, and see what else needs to be changed here.

Reply:
I see. 🤔

Reply:
@symbolist  see 02fa248 for my fix for this issue. I'm still deploying it to the sandbox for testing, but would appreciate your input.

Reply:
The changes make sense to me, thanks.